### PR TITLE
Better logs for gateway [0.2]

### PR DESCRIPTION
**Pull request by Maan2003**
backports #4175 

FYI 5bf8f219c102d37711cd9496308458b107504496 didn't backport trivial

### CODE REVIEW

File: fedimint-core/src/util/mod.rs [hunk 1]
```diff
@@ -217,6 +218,81 @@ pub async fn write_new_async<P: AsRef<Path>, C: AsRef<[u8]>>(
         .await
 }
 
+#[derive(Debug, Clone)]
+pub struct Spanned<T> {
```
Comment (dpc) [inline fedimint-core/src/util/mod.rs:222]: Comment?

Comment (dpc) [reply fedimint-core/src/util/mod.rs:222]: I'm confused what is this trying to accomplish.

Comment (Maan2003) [reply fedimint-core/src/util/mod.rs:222]: i don't want to this backport from master.

it is utility struct to enter a span whenever you want to use the wrapped object

**Review by dpc — APPROVED** [review]

File: fedimint-core/src/util/mod.rs (hunk 1 above)
Comment (dpc) [reply fedimint-core/src/util/mod.rs:222]: Missed opportunity to call it "spanner". :D

**Pull request merged**